core/src/main/java/tachyon/web/UIWebServer.java

```diff
@@ -55,11 +56,14 @@
    * @param serverName Name of the server
    * @param address Address of the server
    * @param masterInfo MasterInfo for the tachyon filesystem this UIWebServer supports
+   * @param conf Tachyon configuration
    */
-  public UIWebServer(String serverName, InetSocketAddress address, MasterInfo masterInfo) {
+  public UIWebServer(String serverName, InetSocketAddress address, MasterInfo masterInfo, 
+      TachyonConf conf) {
+    Preconditions.checkNotNull(conf, "Configuration cannot be null");
```

Comment:
Kind of unrelated to this specific MR, but since preconditions are being introduced, do we want to check the other arguments too?